### ISSUE TITLE
JSON Formatting Ignored or Doesn't Work

### ISSUE THREAD

**Issue by TorreyGarland** [thepirat000/Audit.NET#369]
In the startup file for a .NET 5 Aspnet MVC Core app:

_ = Audit.Core.Configuration.Setup().UseFileLogProvider(config =>
			{
				_ = config.Directory("c:\\temp\\logs");
				_ = config.FilenamePrefix("gtfo-");
				_ = config.JsonSettings(new JsonSerializerSettings
				{
					Formatting = Formatting.Indented
				});
			});

log successfully created, but json indented formatting totally ignored.

**Comment by thepirat000** [comment]
Try with the correct syntax

i.e.:

```c#
Audit.Core.Configuration.Setup().UseFileLogProvider(config => config
    .Directory(@"c:\temp\logs")
    .FilenamePrefix("gtfo-")
    .JsonSettings(new JsonSerializerSettings
    {
        Formatting = Formatting.Indented
    }));
```

**Comment by TorreyGarland** [comment]
copied and pasted the correct syntax.

Output from log files (data from UserName, MachineName, and DomainNameRemoved) => still unformatted/unindented:

{"EventType":"GET /Index","Environment":{"UserName":"","MachineName":"","DomainName":"","CallingMethodName":"WebApplication10.Pages.IndexModel.OnGet()","AssemblyName":"WebApplication10, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null","Culture":"en-US"},"StartDate":"2021-02-14T01:18:07.7457786Z","EndDate":"2021-02-14T01:18:07.7548787Z","Duration":9,"Action":{"TraceId":"80000006-0005-ee00-b63f-84710c7967bb","HttpMethod":"GET","ControllerName":null,"ActionName":"/Index","ViewPath":"/Index","FormVariables":null,"ActionParameters":{},"RequestBody":{"Type":null},"ResponseBody":{"Type":"PageResult"},"UserName":null,"RequestUrl":"https://localhost:44350/","IpAddress":"::1","ResponseStatus":null,"ResponseStatusCode":200,"Headers":{"Accept":"text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9","Accept-Encoding":"gzip, deflate, br","Accept-Language":"en-US,en;q=0.9","Connection":"close","Cookie":".AspNetCore.Antiforgery.rpvwIJNcDhk=CfDJ8Nnp-YtdK3tGuX2GtQWzILw7pAr037TO52YXyaPnt1KPt7eHRYNZNgZQY4d8OgvZUipSwrjfdEKQhZM_0HeLD9kxepjzLCkMkm8v3XS2YbIrNZs12-pJq5PwJp2nFCpQo6mEJPKoVappi42xLi73SA0; .AspNetCore.Antiforgery.dcIV9x6FzQU=CfDJ8Nnp-YtdK3tGuX2GtQWzILzlsWnrPnZdaPLN1AFVuJ6OiduyqvAXITCg_5R1Rz4Zy2MLm6DsVpROAjasaEFdjAcfVQpkEpITIKrKf_AHE98LfNwqj32Mpiq6ESkkqKrCHioqQtav91BZlOYn7A-6iZI","Host":"localhost:44350","User-Agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36 OPR/73.0.3856.344","upgrade-insecure-requests":"1","sec-fetch-site":"none","sec-fetch-mode":"navigate","sec-fetch-user":"?1","sec-fetch-dest":"document"}}}

**Comment by thepirat000** [comment]
could you share all your configuration code? both fot Audit.Core and Audit.EntityFramework

**Comment by thepirat000** [comment]
Oh I see what the problem is... The async implementation of SaveFile is not taking into account the JsonSettings [here](https://github.com/thepirat000/Audit.NET/blob/6a8c0216c23b867143f89f1401faffdd35f70808/src/Audit.NET/Providers/FileDataProvider.cs#L141)

will try to fix this soon

**Comment by thepirat000** [comment]
This was fixed on version 16.4.5, please upgrade your references and re-test.

**Comment by TorreyGarland** [comment]
Thanks
